### PR TITLE
NIFI-12993 Add auto commit feature and add batch processing

### PR DESCRIPTION
**Pull request by jrsteinebrey**
NIFI-12993 PutDatabaseRecord: add auto commit property and fully implement Batch Size for sql statement type

# Summary

[NIFI-12993](https://issues.apache.org/jira/browse/NIFI-12993)
PutDatabaseRecord: add auto commit property and fully implement Batch Size for sql statement type

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [X] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI-12993) issue created

### Pull Request Tracking

- [X] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-12993`
- [X] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-12993`

### Pull Request Formatting

- [X] Pull Request based on current revision of the `main` branch
- [X] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.
I added unit tests to cover many significant combinations of property values

### Build

- [X] Build completed using `mvn clean install -P contrib-check`
  - [X] JDK 21

### Licensing

No new dependencies added

### Documentation

- [X] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java [hunk 1]
```diff
@@ -466,9 +479,41 @@ protected Collection<ValidationResult> customValidate(ValidationContext validati
...
+        if (auto_commit && rollback_on_failure) {
+            validationResults.add(new ValidationResult.Builder()
+                    .subject(RollbackOnFailure.ROLLBACK_ON_FAILURE.getDisplayName())
+                    .explanation(format("'%s' cannot be set to 'true' when '%s' is also set to 'true'. "
+                                    + "Transaction rollbacks for batch updates cannot be supported when auto commit is set to 'true'",
+                            RollbackOnFailure.ROLLBACK_ON_FAILURE.getDisplayName(), AUTO_COMMIT.getDisplayName()))
+                    .build());
+        }
+
+        if (auto_commit && !isMaxBatchSizeHardcodedToZero(validationContext)) {
```
Comment (dan-s1) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java:493]: Variables should use camel case without the use of any underscores
```suggestion
        final boolean autoCommit = validationContext.getProperty(AUTO_COMMIT).asBoolean();
        final boolean rollbackOnFailure = validationContext.getProperty(RollbackOnFailure.ROLLBACK_ON_FAILURE).asBoolean();
        if (autoCommit && rollbackOnFailure) {
            validationResults.add(new ValidationResult.Builder()
                    .subject(RollbackOnFailure.ROLLBACK_ON_FAILURE.getDisplayName())
                    .explanation(format("'%s' cannot be set to 'true' when '%s' is also set to 'true'. "
                                    + "Transaction rollbacks for batch updates cannot be supported when auto commit is set to 'true'",
                            RollbackOnFailure.ROLLBACK_ON_FAILURE.getDisplayName(), AUTO_COMMIT.getDisplayName()))
                    .build());
        }

        if (autoCommit && !isMaxBatchSizeHardcodedToZero(validationContext)) {
```

File: nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java [hunk 2]
```diff
@@ -498,84 +548,98 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
...
+        Connection connection = null;
         boolean originalAutoCommit = false;
         try {
-            final Connection connection = dbcpService.getConnection(flowFile.getAttributes());
-            connectionHolder = Optional.of(connection);
+            connection = dbcpService.getConnection(flowFile.getAttributes());
 
             originalAutoCommit = connection.getAutoCommit();
-            if (originalAutoCommit) {
+            final boolean auto_commit = context.getProperty(AUTO_COMMIT).asBoolean();
```
Comment (dan-s1) [inline nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PutDatabaseRecord.java:558]: Ditto as before use camel case without underscores

**Comment by mattyb149** [comment]
Reviewing...

**Comment by mattyb149** [comment]
I recommend putting all the unit tests in one file and not using abbreviations for the test name.

**Comment by jrsteinebrey** [comment]
@dan-s1 I accidentally hit a button that requested yoiu rereview this PR. That is up to you. I intended to ask for the rereview from mattyb149 and he intends to look at it.

**Pull request closed without merging**